x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsPendingDeletionsIntegTests.java

```diff
@@ -0,0 +1,575 @@
+            try {
+                final String masterNode = internalCluster().getMasterName();
+                blockMasterOnShardLevelSnapshotFile(repository, getRepositoryData(repository).resolveIndexId(index).getId());
+
+                final String cloneTarget = "target-snapshot";
+                final ActionFuture<AcknowledgedResponse> cloneFuture = clusterAdmin().prepareCloneSnapshot(
+                    repository,
+                    snapshot.getName(),
+                    cloneTarget
+                ).setIndices(index).execute();
+                awaitNumberOfSnapshotsInProgress(1);
+                waitForBlock(masterNode, repository);
+
+                assertAcked(client().admin().indices().prepareDelete(mountedIndex(index)));
+                awaitSnapshotPendingDeletion(snapshot);
+                assertFalse(cloneFuture.isDone());
+
+                unblockNode(repository, masterNode);
+                awaitNoMoreSnapshotsDeletions();
+                assertAcked(cloneFuture.get());
+
+                expectThrows(
+                    SnapshotMissingException.class,
+                    () -> client().admin().cluster().prepareGetSnapshots(repository).setSnapshots(snapshot.getName()).get()
+                );
+            } catch (Exception e) {
+                throw new AssertionError(e);
+            }
+        });
+    }
+
+    public void testSearchableSnapshotIsDeletedWithOnGoingDeletion() throws Exception {
+        mountIndexThenExecute((repository, snapshot, index) -> {
+            try {
+                blockMasterOnWriteIndexFile(repository);
+                final String masterNode = internalCluster().getMasterName();
+
+                final ActionFuture<AcknowledgedResponse> deleteFuture = startDeleteSnapshot(repository, snapshot.getName());
+                waitForBlock(masterNode, repository);
+
+                assertAcked(client().admin().indices().prepareDelete(mountedIndex(index)));
+                awaitSnapshotPendingDeletion(snapshot);
+                assertFalse(deleteFuture.isDone());
+
+                unblockNode(repository, masterNode);
+                awaitNoMoreSnapshotsDeletions();
+                assertAcked(deleteFuture.get());
+
+                expectThrows(
+                    SnapshotMissingException.class,
+                    () -> client().admin().cluster().prepareGetSnapshots(repository).setSnapshots(snapshot.getName()).get()
+                );
+            } catch (Exception e) {
+                throw new AssertionError(e);
+            }
+        });
+    }
+
+    public void testSearchableSnapshotsDeletionsWithConcurrentDeletes() throws Exception {
+        final String repository = "repository";
+        final Settings.Builder repositorySettings = randomRepositorySettings();
+        createRepository(repository, FsRepository.TYPE, repositorySettings);
+
+        final List<String> indices = new ArrayList<>();
+        for (int i = 0; i < randomIntBetween(1, 10); i++) {
```

Comment:
This will bias towards smaller lists, since the `randomIntBetween` is evaluated for every loop iteration. I think you intended to make this constant instead? Otherwise it deserves a comment.